<launch>

  <arg name="gui" default="true" />
  <arg name="device" default="-1" />
  <arg name="model_file" default="2024-10-02-grape_ncnn_model" />

  <group ns="left_vzense_camera" >
    <include file="$(find vzense_demo)/launch/object_detection.launch" >
      <arg name="input_image" default="/camera/left_vzense_camera/color/image_raw" />
      <arg name="input_depth_image" default="/camera/left_vzense_camera/transformedDepth/image_raw" />
      <arg name="input_cloud" default="/camera/left_vzense_camera/transformedDepth/points" />
      <arg name="manager" default="/camera/left_vzense_camera/vzense_manager" />
      <arg name="gui" default="$(arg gui)" />
      <arg name="device" default="$(arg device)" />
      <arg name="model_file" default="$(arg model_file)" />
    </include>
  </group>

  <group ns="right_vzense_camera" >
    <include file="$(find vzense_demo)/launch/object_detection.launch" >
      <arg name="input_image" default="/camera/right_vzense_camera/color/image_raw" />
      <arg name="input_depth_image" default="/camera/right_vzense_camera/transformedDepth/image_raw" />
      <arg name="input_cloud" default="/camera/right_vzense_camera/transformedDepth/points" />
      <arg name="manager" default="/camera/right_vzense_camera/vzense_manager" />
      <arg name="gui" default="$(arg gui)" />
      <arg name="device" default="$(arg device)" />
      <arg name="model_file" default="$(arg model_file)" />
    </include>
  </group>

</launch>
